cetmix_tower_yaml/models/cx_tower_yaml_mixin.py

```diff
@@ -110,8 +120,59 @@ def _post_process_record_values(self, values):
             values.update(
                 {"access_level": self.TO_YAML_ACCESS_LEVEL[values["access_level"]]}
             )
+        # Post process m2o fields
+        for key, value in values.items():
+            if key.endswith("_id"):
```

Comment:
in this case there is no guarantee that the specified field will be exactly the relation field, it is necessary to check exactly its type without focusing on how the field is called:

```python

model_fields = self._fields
for field_name, field_value in values.items():
    field = model_fields.get(field_name)
    if field and field["type"] == "many2one":
        values.update({
            field_name: self._process_m2o_value(field, field_value, record_mode=True)
        })

```

Reply:
@GabbasovDinar I agree. However we can imply naming conventions at the project level.
It's also needed because of the YAML-> Odoo conversion. There simply no field types there so the only thing we can rely on is the field suffix.

Reply:
@GabbasovDinar it's also checked [here](https://github.com/cetmix/cetmix-tower/blob/996f9a6f4e541a30b6831d5418783328398f7517/cetmix_tower_yaml/models/cx_tower_yaml_mixin.py#L151)

Reply:
@ivs-cetmix Why not add field type validation as I pointed out above?

Reply:
In this case there is no need to remember about any YAML file, and it is immediately clear from the code that a specific type of field is expected. the code is also cleaner and clearer